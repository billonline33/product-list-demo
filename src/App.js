import React, { Component } from 'react';
import './App.css';
import { Route, withRouter } from 'react-router-dom';
import ProductList from './components/ProductList/ProductList';
import queryString from 'query-string';

export class App extends Component {
  render() {
    const search = this.props.location.search;
    // change URLSearchParams to queryString because
    // create-react-app has Jest v.20 but this version does`t support
    // URLSearchParams, only Jest 22 supports it.
    // but to install Jest 22 i would have to eject create-react-app

    return (
      <div className="App">
        <Route
          path={'/'}
          exact
          component={() => (
            <ProductList queryParams={queryString.parse(search)} />
          )}
        />
      </div>
    );
  }
}

export default withRouter(App);
