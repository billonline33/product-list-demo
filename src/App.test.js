import React from 'react';
import { App } from './App';
import ConnectedProductList from './components/ProductList/ProductList';
import { shallow } from 'enzyme';

describe('App component testing', () => {
  const mockProps = {
    location: {
      search: ''
    }
  };
  const wrapper = shallow(<App {...mockProps} />);

  it('App renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('App does`t render ProductList if not correct route', () => {
    // first we find Route with path = '/' and compare to component
    // if you will set component in this way - component={ProductList}
    // test will be - expect(wrapper.find('Route[exact=true][path="/"]').first().prop('component')).toBe(ConnectedProductList);
    // but you use arrow functions so we have to invoke it and compare to ConnectedProductList with queryParams
    expect(
      wrapper
        .find('Route[exact=true][path="/"]')
        .first()
        .prop('component')()
    ).toEqual(<ConnectedProductList queryParams={{}} />);
  });
});
