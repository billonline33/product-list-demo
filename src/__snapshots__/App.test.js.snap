// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component testing App renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    location={
        Object {
            "search": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Route
        component={[Function]}
        exact={true}
        path="/"
/>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "component": [Function],
        "exact": true,
        "path": "/",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
